hide the owner of a ban in the channel banlist from non-chanops

diff -r 67928388e226 ircd/channel.c
--- a/ircd/channel.c	Sun Jan 11 22:38:39 2009 +0000
+++ b/ircd/channel.c	Sun Jan 11 22:38:40 2009 +0000
@@ -1243,13 +1243,18 @@
 static void send_ban_list(struct Client* cptr, struct Channel* chptr)
 {
   struct Ban* lp;
+  int showbanowner = 0;
 
   assert(0 != cptr);
   assert(0 != chptr);
 
+  /* hide who set the ban from non-chanops */
+  if (IsAnOper(cptr) || is_chan_op(cptr, chptr))
+    showbanowner = 1;
+
   for (lp = chptr->banlist; lp; lp = lp->next)
     send_reply(cptr, RPL_BANLIST, chptr->chname, lp->banstr,
-	       lp->who, lp->when);
+	       showbanowner ? lp->who : "*", lp->when);
 
   send_reply(cptr, RPL_ENDOFBANLIST, chptr->chname);
 }
